lib/content_security_policy/extension.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+class ContentSecurityPolicy
+  module Extension
+    extend self
+
+    def site_setting_extension
+      { script_src: SiteSetting.content_security_policy_script_src.split('|') }
+    end
+
+    def plugin_extensions
+      DiscoursePluginRegistry.csp_extensions
+    end
+
+    THEME_SETTING = 'extend_content_security_policy'
+
+    def theme_extensions
+      cache['theme_extensions'] ||= find_theme_extensions
```

Comment:
I was wondering if I should cache the final combined CSP, but that means a lot more cache invalidation cases.